event/events.go

```diff
@@ -47,12 +54,21 @@ func Multiplex(events ...EventEmitter) *multiplexedEvents {
 
 // The events struct has methods for working with events.
 type events struct {
-	eventSwitch *evts.EventSwitch
+	eventSwitch *go_events.EventSwitch
 }
 
 // Subscribe to an event.
-func (this *events) Subscribe(subId, event string, callback func(evts.EventData)) error {
-	this.eventSwitch.AddListenerForEvent(subId, event, callback)
+func (this *events) Subscribe(subId, event string,
+	callback func(txs.EventData)) error {
+	cb := func(evt go_events.EventData) {
+		eventData, err := mapToOurEventData(evt)
+		if err != nil {
+			panic(fmt.Sprintf("Failed to map go-events EventData to our EventData %v",
+				err))
```

Comment:
I considered something like this, but since go-events doesn't use an erring callback it gets a bit weird with asynchrony. If you really wanted to do it I think using some error channels that you then need to check would be the way. I think we're trying to guard against a 'our coding' problem here rather than something we should really expect at runtime (we control the tendermint version).

Perhaps a nicer middle ground is to `log.Error(...)`. That way it should show up as serious but the whole stack doesn't go down just because someone subscribes to a tendermint event we don't understand..?